doc/RDO.md

```diff
@@ -0,0 +1,48 @@
+Introduction
+--------------
+
+The AV1 bitstream format allows a nearly limitless combination of ways to encode any particular video frame. The purpose of rav1e's RDO loop is to search this incredibly large space of probabilities and arrive at a single solution for each frame.
+
+The parameter rav1e tries to optimize is "RDO score". This is computed as follows:
+
+```score = distortion + rate*lambda```
+
+Distortion is a metric selected based on user options, such as MSE for PSNR optimized mode or cdef-dist for psychovisual optimized mode. Rate is the number of bits used in the bitstream (commonly in 1/8 bit units throughout the codebase). Lambda is the current operating point (tradeoff between higher quality or lower bitrate), and is directly related to the quantizer setting.
+
+![](tile_group.svg)
+
+The bitstream is composed of a hiearchy of units, and rav1e's RDO loops are structured in the same way. After computing some lookahead information, rav1e's first task is to split the frame into tiles (sometimes just one). These are processed completely independently, and usually in parallel.
+
+The next split is into superblocks, which are currently always 64x64 pixels in the luma plane. rav1e processes superblocks serially, one at a time. Although choices made in a superblock affect later superblocks, rav1e will never revisit a superblock once it has determined the best way to code it - it writes it and continues on.
+
+The next split is into partitions. Partitions are actually a tree structure, so the search here is recursive - rav1e has to compare treating the current level as a leaf node versus recursing further, all the way down the tree. To keep the number of permutations manageable, rav1e (in all speed levels but 0) traverses the tree in a top-down approach, and will stop splitting if the RDO score does not improve from the previous split. This can still potentially mean a large number of total combinations, though, so higher speed levels have a limited range of levels to search.
```

Comment:
"(in all speed levels but 0)" ==> "(in all speed levels but 0 and 1)", since currently speed 0 and 1 are using bottom-up partition search.